data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -106,7 +107,7 @@ public void doInitialize() {
     try {
         doInitializeInternal();
     } catch (IOException e) {
-        LOG.warn("Failed to initialize OpenSearch sink, retrying. Error - " + e.getCause());
+        LOG.warn("Failed to initialize OpenSearch sink, retrying. Error - " + (Objects.isNull(e.getCause()) ? e : e.getCause()));
```

Comment:
Please use:

```
 LOG.warn("Failed to initialize OpenSearch sink, retrying. Error - {}", e.getCause());
```

It will handle nulls, simplify your code, and may avoid the concat when logging is disabled.

Reply:
Also, do we eventually include the stack trace if it fails in the final try?

If not, we should probably log this _as a full exception_.

```
LOG.warn("Failed to initialize OpenSearch sink, retrying.", e.getCause());
```

This will include the full stack trace to help track the issue.